Snowflake.Data/Configuration/EasyLoggingConfigFinder.cs

```diff
@@ -43,24 +46,34 @@ private string GetFilePathEnvironmentVariable()
             return GetFilePathFromInputParameter(filePath);
         }
         
-        private string GetFilePathFromTempDirectory() => SearchForConfigInDirectory(Path.GetTempPath());
+        private string GetFilePathFromTempDirectory() => SearchForConfigInDirectory(Path.GetTempPath, "temp");
 
-        private string GetFilePathFromHomeDirectory() => SearchForConfigInDirectory(GetHomeDirectory());
+        private string GetFilePathFromHomeDirectory() => SearchForConfigInDirectory(GetHomeDirectory, "home");
         
         private string GetFilePathFromInputParameter(string filePath) => string.IsNullOrEmpty(filePath) ? null : filePath;
 
         private string GetHomeDirectory() =>_environmentOperations.GetFolderPath(Environment.SpecialFolder.UserProfile);
 
-        private string GetFilePathFromDriverLocation() => SearchForConfigInDirectory(".");
+        private string GetFilePathFromDriverLocation() => SearchForConfigInDirectory(() => ".", "driver");
 
-        private string SearchForConfigInDirectory(string directory)
+        private string SearchForConfigInDirectory(Func<string> directoryProvider, string directoryDescription)
```

Comment:
Do we really need this `directoryDescription`? Maybe we can move `var directory = ...` outside of `try` and just use it in `catch`? It would be more even meaningful.

Reply:
and what if the operation of getting the directory fails?
This way I did it I handle all sort of errors from different reasons with little code changes so I like it better.